### PR TITLE
EN-50493 - Ubuntu focal base and language images

### PR DESCRIPTION
**Pull request by jsondheimer**

### CODE REVIEW

File: base-focal/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Any container built on top of the socrata/base-xenial image will default to running the whatever script is dropped in `/etc/ship.d/run`. Alternatively, you can invoke any script dropped in /etc/ship.d/ or any executable in the container via the CMD argument:
+
+```bash
+    $ docker pull socrata/base-bionic
+    $ docker run --rm -t -i socrata/base-bionic [CMD]
+
+    # Examples:
+    $ docker run --rm -t -i socrata/base-bionic # runs /etc/ship.d/run in the container
+    $ docker run --rm -t -i socrata/base-bionic bash        # launch a bash shell (on PATH)
```
Comment (dosieckityler) [inline base-focal/README.md:30]: A few more cases where we say bionic/xenial and maybe mean focal?  Do we even need these docs?

File: base-focal/clortho-get [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/ruby
```
Comment (dosieckityler) [inline base-focal/clortho-get:1]: These poor clortho scripts are duplicated in like half a dozen places already.  I wish we could DRY these out . . .

File: base-focal/set_ark_host [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
```
Comment (dosieckityler) [inline base-focal/set_ark_host:1]: Same thoughts about DRYing out . . .

File: java8-focal/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+socrata/java8-bionic
```
Comment (dosieckityler) [inline java8-focal/README.md:1]: s/bionic/focal/

File: python3-focal/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+socrata/python-binoic
```
Comment (dosieckityler) [inline python3-focal/README.md:1]: s/bionic/focal/

File: ruby-focal/README.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+# Launch shell in the container
+docker run --rm -t -i socrata/ruby-bionic:<version> bash
+
+# Bind mount a directory into the container and build or run something
+docker run --rm -t -i -v `pwd`:/opt/my_app socrata/ruby-bionic:<version> ruby my_app.rb
+```
+
+## Available versions
+
+- `socrata/ruby-bionic:2.7.x`
```
Comment (dosieckityler) [inline ruby-focal/README.md:22]: Could we boldly deprecate 2.7 or 3.0 in focal and force people to up to 3.1 if they are already upgrading Ubuntu anyhow?  Let's not even give them the option of laggin behind.  Want focal?  Get on 3.1

File: runit-ruby-focal/README.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    $ docker run --rm -t -i socrata/runit-ruby-bionic:<version> bash
+
+    # Bind mount a directory into the container and build or run something
+    $ docker run --rm -t -i -v `pwd`:/opt/my_app socrata/runit-ruby-bionic:<version> ruby my_app.rb
+
+
+### Available versions
+
+- `socrata/runit-ruby-bionic` _alias for `socrata/runit-ruby-bionic:2.7.x`_
+- `socrata/runit-ruby-bionic:2.7.x`
```
Comment (dosieckityler) [inline runit-ruby-focal/README.md:22]: Same thoughts as above about ditching 2.x and 3.0.

**Review by dosieckityler — CHANGES_REQUESTED** [review]

File: base-focal/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+RUN apt-get -y update && \
+    apt-get -y upgrade; \
+    apt-get -y install \
+          build-essential \
+          locales \
+          curl \
+          dnsutils \
+          zip \
+          ruby2.7 \
```
[the remark this answers is not in the document]
Comment (jsondheimer) [reply base-focal/Dockerfile:20]: This is just using the packages provided via ubuntu's repos. Ruby 2.7 is all they offer for focal.

File: runit-java8-focal/README.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
+socrata/runit-bionic
```
Comment (dosieckityler) [inline runit-java8-focal/README.md:1]: s/bionic/focal/

File: runit-java8-focal/README.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+====================
+
+Image based on phusion/baseimage-docker that establises a base set of patterns and tools for building other containers with support for multiple processes via runit. This image is packaged with OpenJDK *version 8* installed.
+
+### Features
+
+- Ubuntu 18.04 LTS. The base system.
```
Comment (dosieckityler) [inline runit-java8-focal/README.md:8]: uhhh, s/18.04/22.04/ I think?

File: ruby-focal/3.1.x/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,41 @@
+FROM socrata/base-bionic
```
Comment (dosieckityler) [inline ruby-focal/3.1.x/Dockerfile:1]: s/bionic/focal/

File: runit-java8-focal/README.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+Inside a docker container, the host’s ip and name are not available.  By invoking these scripts, we can make the container aware of its host as necessary.  They create the environment variables ARK_HOST and ARK_HOSTNAME.
+
+- env_parse:
+A tool for generating config files from [jinja](http://jinja.pocoo.org/) templates and environment variables.
+
+This script lets a service owner build configuration files that are created at run time from environment variables and a template file.  The template language is jinja and any template variables must be available as e  nvironment variables.  env_parse takes one argument, the template, and an optional argument for the output file.  If omitted, the output file is identical to the template file name with the trailing .j2 removed.
+
+### Usage
+
+Any container built on top of the socrata/runit-bionic image will default to running the whatever services are configured in /etc/service according via runit.
```
Comment (dosieckityler) [inline runit-java8-focal/README.md:34]: s/bionic/focal/

File: runit-java8-focal/README.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+Anything placed in /etc/my_init.d will be run on startup in lexigraphical order before runit is invoked. A non-zero return code from any of these will halt the container.
+
+### Example
+
+Assuming we build an image called awesome_sauce from a Dockerfile like this:
+
+    FROM socrata/runit-bionic
```
Comment (dosieckityler) [inline runit-java8-focal/README.md:42]: s/bionic/focal/

**Review by dosieckityler — CHANGES_REQUESTED** [review]

**Review by dosieckityler — APPROVED** [review]

**Pull request merged**